src/App.js

```diff
@@ -60,7 +70,18 @@ class App extends Component {
     const apiKey = this.getApiKey()
 
     if (!apiKey && this.isTableau) {
-      this.redirectToAuth()
+      if (code) {
+        getToken(code)
+        .then(response => {
+          const token = response.data.access_token
+          if (token) {
+            window.localStorage.removeItem('DW-CODE-VERIFIER')
+            window.location = `${process.env.REACT_APP_OAUTH_ROOT_URL}?token=${token}`
```

Comment:
It shouldn't be necessary to change the `location`, just store the token returned by the API.  You may need to rearrange some of the code in the constructor.

Reply:
It is.

At this point the page is on the redirect URL (`/callback?code={auth code}`) and needs to be redirected back to the home URL.

Reply:
Ah, ok.  Is that arbitrary though?  You could set the redirect to just the domain since there is no routing in the application.  Something to keep in mind for a future enhancement.